test/parallel/test-tls-connect-allow-half-open-option.js

```diff
@@ -19,3 +20,41 @@ const tls = require('tls');
   const socket = tls.connect({ allowHalfOpen: false, lookup() {} });
   assert.strictEqual(socket.allowHalfOpen, false);
 }
+
+
+const server = tls.createServer({
+  key: fixtures.readKey('agent1-key.pem'),
+  cert: fixtures.readKey('agent1-cert.pem'),
+}, common.mustCall((socket) => {
+  server.close();
+
+  socket.setEncoding('utf8');
+  socket.on('data', common.mustCall((chunk) => {
+    assert.strictEqual(chunk, 'Hello');
```

Comment:
I was a bit thrown by the conditional, it seemed to me the test should know exactly whether `socket.writable` was true or false, but then I read the rest of the  test and I see why. I suggest changing this to something like below:
```
if (chunk === 'Hello') {
socket.end(chunk);
} else {
assert(chunk === 'Bye');
assert(!socket.writable);
}
```